### ISSUE TITLE
Incorrect Instruction Lab07

### ISSUE THREAD

**Issue by kimpatrickdelosreyes** [MicrosoftLearning/20532-DevelopingMicrosoftAzureSolutions#246]
# Module: 07
#### Step: 00

Description of issue
Exercise 2, Task 2
return new DownloadPayload { Stream = blobStream, ContentType = blob.Properties.ContentType };
Should be
return blob;

**Comment by seesharprun** [comment]
Just giving a heads-up that I don't know if anyone is maintaining this course anymore.

**Comment by molenaar** [comment]
I hope so, because these are the only labs we have for Azure development. The az-20x courses still don’t have any labs.

**Comment by ConcreteGannet** [comment]
Yes. The Cloud Workshop courses 405xx are now freely available on GitHub - start at https://github.com/Microsoft/MCW . Some of those might help. I am hopeful the new AZ-203 due in April will have better labs.

**Comment by seesharprun** [comment]
[Speaking for myself as a MCT]

That is weird. I was on the authoring team for az-200 and az-201 and there were definitely labs at the end of each course. There was a very limited time to write them so they weren't as good as 20532 in my opinion. I wonder if the labs didn't make it to release.

I'm also on the authoring team for az-203. I'm limited in what I can say but I can tell you that the authoring team built out some labs that I think are better than the 20532 and 20535 labs. I'm really excited for people to see them this Spring. I don't know where they will be hosted and I won't be managing them so I look forward to submitting issues, marking PRs and being part of the conversation with everyone else this spring.

In short, this course and the GitHub repo are done.

**Comment by ConcreteGannet** [comment]
Thanks Sidney.

If there were formal labs at the end of each course or each module, no, they did not make it to release. The courses are on the Learning Download Center, so you can see what is available to us. Would it be possible to put your labs as they stand on Github?

There are some hands on activities in AZ-200 and AZ-201. But they are part of the normal flow of the course, and not in a clearly separated section. So it is not immediately obvious if what was intended was an outline of steps, a demonstration, or a lab. Because they are integrated with the course material, we can't maintain them in GitHub. The software configuration required for the activities to work is only vaguely specified, and there are no VMs, no starters and no suggested solutions. I would argue formal labs are important to customers and a differentiator between a MOC course and a quickstart.

Some topics have no lab, just a QR code for a demonstration video. If you are in a training room with a desktop PC, the QR code is near useless, and there isn't a title or shortened URL beside the QR code
to help you find the video in question.

And while I'm at it, AZ-200 and 201 are the first Microsoft courses I have seen in 22 years that do not have the copyright and legal stuff at the front.

Lab instructions on GitHub is one of the best things to happen in the Microsoft training world in the last few years, and it's a retrograde step for them to be missing from these courses.

I understand you can't say much about a course in the pipeline. Can you answer this ... I have seen a release date of 1 April for AZ-203. Do you think that will be about right?

**Comment by denschork** [comment]
Why is anyone (MCT’s) prepping these courses anymore when NOBODY is signing up for them?

MOC demand is essentially ZERO and membership in the MCT program has become essentially worthless.

How many MCT’s are actually making money delivering MOC?

**Comment by tiagocostapt** [comment]
@denschork that is a opinion I guess almost 10.000 MCTs that are in the program are just spending money. I'm a MCT for 15 years and I can't complain about lack of work, I have just the opposite of it. Can't handle all the requests. I see the MCT certification not just something directly connect with the permission to deliver MOC training, but it opens all sort of other doors to other experiences. Again this is just an opinion, in the case mine.

**Comment by denschork** [comment]
<<. I'm a MCT for 15 years and I can't complain about lack of work, I have just the opposite of it.>>

You are not getting MOC classes. Open source, yes. But MOC is dead. MSFT has killed the ILT MOC.

I’ve been an MCT over 22 years and I’m connected with over 30 training centers. NONE of them are doing MOC anymore on a regular basis. There is NO DEMAND. Killed by a combo of switch to open source and MSFT giving training away.

**Comment by seesharprun** [comment]
I'm limited in what I can say but I can share a little bit:

Handing off content to Microsoft is sometimes a black hole. As you saw with AZ200 and AZ201, labs were built but they were integrated into the course as something else that wasn't a formal lab with a VM. By the way, there are MOC courses out there without a VM. 20532 didn't get a VM until the latest revision.

I expect much of the same with AZ203, the authoring team will handoff to Microsoft at some point and it's up to them to decide on the exact release date. This is true with any vendor and any MOC course. I expect the AZ203 labs to live on GitHub based on my experience with some of the more recent AZ courses. I don't know who will maintain them or anything beyond that.

**[Warning Opinion Section Incoming]**

Regarding the MCT program, learning is changing. The industry has clearly seen that learning takes many modalities (self-paced online, ILT, video, etc). ILT is still huge and can command great rates. Packaged learning, however, is no longer the default for ILT. Many clients are paying for variations on packaged courses or custom training. It's impossible to expect any cloud company to build a separate packaged course for each cloud service or one course to cover them all and this has forced the industry to shift. You may see this shift yourself where the demand for training is increasing but the desire to use off-the-shelf courses is dropping.

It's a wild ride we are on but this is definitely not the end of ILT. I make a living now by blending custom content with select modules from MOC so I can get the "best of both worlds". I'm able to charge a premium because there's a lot of people out there who won't do that. I have a Fortune 500 client who regularly requests Azure training and I can't say "no, we don't have a course for that yet". My answer is always yes and I quote them a sliding rate based on the percentage of custom content and reusable content.

**Comment by seesharprun** [comment]
Unfortunately, this project is no longer actively maintained.

**Comment by denschork** [comment]
Unfortunately,MCT program is no longer maintained.
